src/cluster/troubleshooting.rst

```diff
@@ -0,0 +1,114 @@
+.. License for the specific language governing permissions and limitations under
+.. the License.
+
+.. _cluster/troubleshooting:
+
+=============================================
+Troubleshooting CouchDB 3 with Weather Report
+=============================================
+
+.. _cluster/troubleshooting/overview:
+
+Overview
+========
+
+``weatherreport`` is an escript and set of tools that diagnoses common
+problems which could affect a CouchDB version 3 node or cluster. It
+does not support version 4 or later.
+
+Here is a basic example of using ``weatherreport`` followed immediately
+by the command's output:
+
+.. code-block:: bash
+
+    $ weatherreport --etc /path/to/etc
+    [warning] Cluster member node3@127.0.0.1 is not connected to this node. Please check whether it is down.
+
+.. _cluster/troubleshooting/usage:
+
+Usage
+=====
+
+For most cases, you can just run the ``weatherreport`` command as
+shown above.  However, sometimes you might want to know some extra
+detail, or run only specific checks. For that, there are command-line
+options. Execute ``weatherreport --help`` to learn more about these
+options:
+
+.. code-block:: bash
+
+    $ weatherreport --help
+    Usage: weatherreport [-c <path>] [-d <level>] [-e] [-h] [-l] [check_name ...]
+
+      -c, --etc                 Path to the CouchDB configuration directory
+      -d, --level               Minimum message severity level (default: notice)
+      -l, --list                Describe available diagnostic tasks
+      -e, --expert              Perform more detailed diagnostics
+      -h, --help                Display help/usage
+      check_name                A specific check to run
+
+To get an idea of what checks will be run, use the `--list` option:
+
+.. code-block:: bash
+
+    $ weatherreport --list
+    Available diagnostic checks:
+
+      custodian            Shard safety/liveness checks
+      disk                 Data directory permissions and atime
+      internal_replication Check the number of pending internal replication jobs
+      ioq                  Check the total number of active IOQ requests
+      mem3_sync            Check there is a registered mem3_sync process
+      membership           Cluster membership validity
+      memory_use           Measure memory usage
+      message_queues       Check for processes with large mailboxes
+      node_stats           Check useful erlang statistics for diagnostics
+      nodes_connected      Cluster node liveness
+      process_calls        Check for large numbers of processes with the same current/initial call
+      process_memory       Check for processes with high memory usage
+      safe_to_rebuild      Check whether the node can safely be taken out of service
+      search               Check the local search node is responsive
+      tcp_queues           Measure the length of tcp queues in the kernel
+
+If you want all the gory details about what WeatherReport is doing,
```

Comment:
I notice we alternately refer to this as "Weather Report", "WeatherReport" and "``weatherreport``".  Do we want to standardize on one spelling in normal prose?

Reply:
We're using `weatherreport` to refer to the actual command line executable, but you're right about the other inconsistency. I lean toward using WeatherReport as the name, but don't have a strong preference either.